isisdataeval Updates, fixes and additional documentation

KrisBecker opened the pull request:
Apologies I missed the merge of #5111 with a few, but perhaps helpful, updates.

isisdata_mockup.py - Added —tojson and —hasher parameters to improve utility

make_isisdata_mockup.sh - Fixed path to test data and added new parameters to the isisdata_mockup.py introduced in this PR

README.md - improved/clarified/updated documentation

<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail including motivation and any context -->

## Related Issue
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->
#5111 

## How Has This Been Validated?
<!--- All changes need to be validated to confirm that they produce -->
<!--- scientifically accurate results. -->
<!--- If your changes include any new algorithms or changes to existing algorithms, -->
<!--- please indicate any publications or references for them. -->
<!--- If you manually validated the changes please indicate -->
<!--- what data you used and how you checked the results. -->

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Infrastructure change (changes to things like CI or the build system that do not impact users)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!--- - [ ] My code follows the code style of this project. -->
- [x] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [x] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [ ] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [ ] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

KrisBecker committed:
isisdataeval fixes, improvements, updates and docs

isisdata_mockup.py - Added —tojson and —hasher parameters to improve utility

make_isisdata_mockup.sh - Fixed path to test data and added new parameters to the isisdata_mockup.py introduced in this PR

README.md - improved/clarified/updated documentation

KrisBecker committed:
CHANGELOG.md correction and update

The CHANGELOG.md was incorrectly merged as the entry for this app was added in the wrong section, which seem wierd. This commit corrects and updates the log.

KrisBecker commented:
Question: Is it recommended/possible to install `isisdata_mockup.py` in a bin-type directory or someplace where its easier to access/use?

Kelvinrr commented on isis/src/system/apps/isisdataeval/tools/README.md:
```diff
@@ -17,29 +17,31 @@ The ISISDATA mockup system is a complete copy of every directory and every file
 optional arguments:
-  -h, --help           show this help message and exit
-  --isisdata ISISDATA  ISISDATA directory root
-  --outpath OUTPATH    NEW ISISDATA directory to create
-  --ghostdir GHOSTDIR  Replaces the --isisdata path with this string to ghost the actual input
-                       directory path
-  --saveconfig, -s     Retain *.db, *.conf files rather than replace with processs info
-  --dryrun, -n         Only print actions but do not execute
-  --verbose, -v        Verbose output
+  -h, --help            show this help message and exit
+  --isisdata ISISDATA   ISISDATA directory root
+  --outpath OUTPATH     NEW ISISDATA directory to create
+  --ghostdir GHOSTDIR   Replaces the --isisdata path with this string to ghost the actual input directory path
+  --hasher {md5,sha1,sha256,all}
+                        Add desired/supported hash algorithms to compute
+  --saveconfig, -s      Retain *.db, *.conf files rather than replace with processs info
+  --tojson TOJSON       Write all ISISDATA file information to output file in JSON format
+  --dryrun, -n          Only print --outpath actions but do not execute
+  --verbose, -v         Verbose output
```
could we just avoid putting in-the-weeds parameter details in the readme and instead instruct to run the command? this way we don't have to keep this in sync.

KrisBecker replied:
Sure make sense. I removed the help and added additional context.

Kelvinrr replied:
Looks like the full output of the help message is still there?

KrisBecker replied:
This section was removed/updated in [this push](https://github.com/DOI-USGS/ISIS3/pull/5163/commits/498c9f4836ee882efe76a9dc9570926341c5eece) on or about April 3. You can also check the [source](https://github.com/KrisBecker/ISIS3/blob/feature/isisdataeval-update/isis/src/system/apps/isisdataeval/tools/README.md) to see the changes.

Kelvinrr replied:
weird, idk why Github it doing me like this.

Kelvinrr commented on isis/src/system/apps/isisdataeval/tools/README.md:
````diff
@@ -63,53 +70,337 @@ Finally, by setting --ghostdir '$ISISDATA', this now provides a connection to
+isisdata_mockup.py --hasher md5 --isisdata /opt/isis3/data/base/dems --ghostdir '$ISISDATA/base/dems' --tojson isisdata_base_dem_md5.json
+```
 
+And here is the contents of the output JSON file, `isisdata_base_dem_md5.json`:
+```
+{
+  "program": {
+    "name": "isisdata_mockup.py",
+    "version": "0.2",
+    "date": "2023-03-29",
+    "runtime": "2023-03-29T21:02:03.721127 UTC",
+    "endtime": "2023-03-29T21:02:39.887436 UTC",
+    "elapsedtime": "0:00:36.166309",
````
could we replace this with a truncated version so it isn't as verbose?

KrisBecker replied:
Done.

Kelvinrr reviewed (commented):
some small comments. 

Partly also my bad for being a little hasty with wanting to merge that PR.

KrisBecker committed:
isisdataeval documentation update via code review

KrisBecker commented:
Any suggestions on where/how to install isisdata_mockup.py? This should be conveniently available to ISIS users if you decide to use this script as mechanism to validate ISISDATA installations.

Kelvinrr commented:
@KrisBecker at the end of the CMakeLists.txt, you can add a line to install it to $INSTALL_PREFIX/bin/ with the .py stripped so we can run it as a command. That's how we've usually done these utility scripts we want common accesss to.

KrisBecker commented:
Ok, so should I scrub all the docs to refer to the renamed extension-less version of the script? At least in the READMEs and the tests scripts?

KrisBecker committed:
isisdataeval install $ISISROOT/bin/isisdata_mockup

isisdata_mockup.py - Install this script in $ISISROOT/bin as isisdata_mockup (no extension)

make_isisdata_mockup.sh - Renamed isisdata_mockup.py to isisdata_mockup

CMakeLists.txt - Add install command for isisdata_mockup

KrisBecker committed:
isisdataeval mockup script minor docs change

KrisBecker commented:
Commits install isisdata_mockup in $ISISROOT/bin (its not in `build/bin`). Had to finesse the permissions.

I also changed the test data mocking script to call this version.

This does create an issue for most of the documentation. It may be easier to just rename the script without the extension and change all the documentation appropriately otherwise I am concerned trying to retain the extension will lead to confusion.

Kelvinrr commented:
That would be helpful, yes.

KrisBecker committed:
Rename isisdata_mockup.py to isisdata_mockup

To ease use and minimize confusion, renamed isisdata_mockup.py to isisdata_mockup (w/o the .py extension)

Updated documentation and install to reflect the change

KrisBecker commented:
Ok, renamed isisdata_mockup and modified docs/scripts to reflect the change.

This PR should be good to go now.

KrisBecker committed:
Merge branch 'DOI-USGS:dev' into feature/isisdataeval-update

KrisBecker commented:
Please let me know if there is anything else I need to do in order to get this PR merged.

Thank you.

Kelvinrr commented:
Once merge conflicts are resolved I'll merge this.

KrisBecker committed:
Merge remote-tracking branch 'origin/dev' into feature/isisdataeval-update

# Conflicts:
#	CHANGELOG.md

KrisBecker committed:
Resolved merge conflict and add CHANGELOG.md entry

This push resolves a merge conflict in the CHANGELOG.md

Added a entry to CHANGELOG.md to describe the PR changes.

KrisBecker committed:
Merge branch 'feature/isisdataeval-update' of https://github.com/KrisBecker/ISIS3 into feature/isisdataeval-update

KrisBecker commented:
Merge conflict resolved in CHANGELOG.md and added an entry with these updates.

KrisBecker committed:
Removed hardcoded path in isisdata_mockup

Kelvinrr commented:
There is a random error after the merge conflicts were fixed in the test. 

```
The following tests FAILED:
	1528 - DemCube.FunctionalTestSpiceinitWebAndShapeModel (Failed)
```

Might be a fluke so I am rerunning Jenkins for now.

Kelvinrr commented:
the error is still there, it did show up after the last commit. But it is a Spiceinit error trying to hit the web server. Might have been down at the time the test ran coincidentally. I'll run one more time, if it still fails on that test it will probably have to be debugged.

Kelvinrr reviewed (approved):
aaaaaay it passed.
